### DISCUSSION TITLE
Redirecting based on previous route?

### DISCUSSION THREAD

**Discussion by SteveDowsett** [TanStack/router#1706] [Q&A]
Hey all, i have a route that id like to some how conditionally redirect to another if the page is being hit fresh, i.e not being navigated to as part of the usual flow and doesnt have a previous route.

How can i determine that? I was thinking the logic would need to go in the beforeLoad for the route, but not sure how i can check what the previous/from route is?

**Comment by SeanCassiere** [comment]
You can set up an event listener using `router.subscribe` and write the previous URL to `window.sessionStorage`.

From there on, its a matter of reading the sessionStorage to get the previous URL and determine what you want to do with that.

**Reply by SteveDowsett** [reply]
Is there no way to determine that in the beforeLoad of the route? i.e does the router not know somehow what the previous route was internally without having to roll my own way of persisting?
